replace klee_alias_function() with FunctionAliasPass

jbuening opened the pull request:
In this PR, I propose a replacement for `klee_alias_function()` following the suggestion by @ccadar in #989. I also incorporated the functionality that was suggested by @SolalPirelli in #818.

To do this, I introduced a new `FunctionAliasPass` that runs during `optimiseAndPrepare()`. I chose this position as it most closely resembles the view on the module that `klee_alias_function()` had before. The pass injects `GlobalAlias`es according to a comand line option, which makes it completely transparent to `Executor` or `ExecutionState`.

Instead of `klee_alias_function("foo", "bar");` there is `-function-alias=foo:bar`. The option `-function-alias` can be used multiple times, the pass will apply the aliases one after the other. Regular expressions as pattern can be used like this: `-function-alias=foo.*:bar`. The pass also checks if all involved functions are present and if their types match. See my test files for some more examples.

~(For the test `BITCAST-ALIAS` to pass, this PR requires #1003, whose patch I therefore included in this branch. I will rebase once the other PR is merged.)~

ccadar commented:
@jbuening Thanks for this excellent contribution, and sorry for not reviewing it on time.  I now did so, and it looks great.  The only necessary changes are to (1) rebase it on master and (2) place the new option in a category; I think either the module-related options or the startup options would fit, what do you think?

jbuening commented:
@ccadar Thanks for taking the time to review this!
I just rebased on master, and regarding your second point: Tough call, but I guess the module option category fits a bit better, so I added it there for now.

ccadar commented:
Great, thanks again, @jbuening !
